app/scripts/controllers/swaps.ts

```diff
@@ -831,6 +717,227 @@ export default class SwapsController {
+    });
+  }
+
+  public setSwapsErrorKey(errorKey: string) {
+    this.update((_state) => {
+      _state.swapsState.errorKey = errorKey;
+    });
+  }
+
+  public setSwapsLiveness(swapsLiveness: { swapsFeatureIsLive: boolean }) {
+    const { swapsFeatureIsLive } = swapsLiveness;
+    this.update((_state) => {
+      _state.swapsState.swapsFeatureIsLive = swapsFeatureIsLive;
+    });
+  }
+
+  public setSwapsQuotesPollingLimitEnabled(quotesPollingLimitEnabled: boolean) {
+    this.update((_state) => {
+      _state.swapsState.quotesPollingLimitEnabled = quotesPollingLimitEnabled;
+    });
+  }
+
+  public setSwapsTokens(tokens: string[]) {
+    this.update((_state) => {
+      _state.swapsState.tokens = tokens;
+    });
+  }
+
+  public setSwapsTxGasLimit(customMaxGas: string) {
+    this.update((_state) => {
+      _state.swapsState.customMaxGas = customMaxGas;
+    });
+  }
+
+  public setSwapsTxGasPrice(customGasPrice: string | null) {
+    this.update((_state) => {
+      _state.swapsState.customGasPrice = customGasPrice;
+    });
+  }
+
+  public setSwapsTxMaxFeePerGas(customMaxFeePerGas: string | null) {
+    this.update((_state) => {
+      _state.swapsState.customMaxFeePerGas = customMaxFeePerGas;
+    });
+  }
+
+  public setSwapsTxMaxFeePriorityPerGas(
+    customMaxPriorityFeePerGas: string | null,
+  ) {
+    this.update((_state) => {
+      _state.swapsState.customMaxPriorityFeePerGas = customMaxPriorityFeePerGas;
+    });
+  }
+
+  public setSwapsUserFeeLevel(swapsUserFeeLevel: string) {
+    this.update((_state) => {
+      _state.swapsState.swapsUserFeeLevel = swapsUserFeeLevel;
+    });
+  }
+
+  public setTradeTxId(tradeTxId: string | null) {
+    this.update((_state) => {
+      _state.swapsState.tradeTxId = tradeTxId;
+    });
+  }
+
+  /**
+   * Once quotes are fetched, we poll for new ones to keep the quotes up to date.
+   * Market and aggregator contract conditions can change fast enough that quotes
+   * will no longer be available after 1 or 2 minutes. When `fetchAndSetQuotes` is
+   * first called, it receives fetch parameters that are stored in state. These stored
+   * parameters are used on subsequent calls made during polling.
+   *
+   * Note: We stop polling after 3 requests, until new quotes are explicitly asked for.
+   * The logic that enforces that maximum is in the body of `fetchAndSetQuotes`.
+   */
+  public stopPollingForQuotes() {
+    if (this.#pollingTimeout) {
+      clearTimeout(this.#pollingTimeout);
+    }
+  }
+
+  /**
+   * This method is used to update the state of the controller for testing purposes.
+   * DO NOT USE OUTSIDE OF TESTING
+   *
+   * @param newState - The new state to set
+   */
+  public __test__updateState = (newState: Partial<SwapsControllerState>) => {
+    this.update((oldState) => {
+      return { swapsState: { ...oldState.swapsState, ...newState.swapsState } };
+    });
+  };
+
+  // Private Methods
+  private async _fetchSwapsNetworkConfig(chainId: ChainId) {
```

Comment:
nit: Could we rename all private properties to consistently follow the hash-prefix naming scheme? 

The hash prefix [is enforced by JavaScript at runtime on the syntax level](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties), whereas the scope keywords `public`/`protected`/`private` are compile-time syntactic sugar that aren't enforced (we also find it unnecessary to use the `public` keyword for this reason).

```suggestion
  // Private Methods
  async #fetchSwapsNetworkConfig(chainId: ChainId) {
```

Reply:
I considered this during the refactor, but the problem we have is that we need to spy on "internal" methods for testing purposes. So I cannot completely enforce privacy at runtime (at least not at the moment). 

The best compromise i found was to label methods as private or public so we can check this with TypeScript. 

Ideally we should redesign the controller data flow so we care only about public methods output and test only that. But for now this involves work that goes a bit out of the scope of this migration to `BaseController` V2

Reply:
Oh you're right looks like there are quite a few of those `spyOn` calls. I'm good with refactoring the tests in a separate ticket.